2003-09-28  Dmitry V. Levin <ldv@altlinux.org>

	* apt-pkg/rpm/rpmpm.cc(pkgRPMPM::ExecRPM):
	Honor "quiet" config option.
	Pass "fancypercent" option.

--- apt-0.5.5cnc4.1/apt-pkg/rpm/rpmpm.cc.orig	2003-09-28 14:18:22 +0400
+++ apt-0.5.5cnc4.1/apt-pkg/rpm/rpmpm.cc	2003-09-28 14:22:42 +0400
@@ -227,6 +227,7 @@ bool pkgRPMPM::ExecRPM(Item::RPMOps op, 
    const char *operation;
    unsigned int n = 0;
    bool Interactive = _config->FindB("RPM::Interactive",true);
+   int quiet = _config->FindI("quiet",0);
    
    Args[n++] = _config->Find("Dir::Bin::rpm","rpm").c_str();
 
@@ -235,7 +236,7 @@ bool pkgRPMPM::ExecRPM(Item::RPMOps op, 
    switch (op)
    {
       case Item::RPMInstall:
-	 if (Interactive)
+	 if (Interactive && quiet < 2)
 	    operation = "-ivh";
 	 else
 	    operation = "-iv";
@@ -243,7 +244,7 @@ bool pkgRPMPM::ExecRPM(Item::RPMOps op, 
 	 break;
 
       case Item::RPMUpgrade:
-	 if (Interactive)
+	 if (Interactive && quiet < 2)
 	    operation = "-Uvh";
 	 else
 	    operation = "-Uv";
@@ -256,8 +257,10 @@ bool pkgRPMPM::ExecRPM(Item::RPMOps op, 
    }
    Args[n++] = operation;
 
-   if (Interactive == false && op != Item::RPMErase)
+   if (Interactive == false && quiet == 0 && op != Item::RPMErase)
       Args[n++] = "--percent";
+   if (Interactive && quiet == 0)
+      Args[n++] = "--fancypercent";
     
    string rootdir = _config->Find("RPM::RootDir", "");
    if (!rootdir.empty()) 
